zookeeper/job.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+from zookeeper import Component
+from inspect import getfullargspec, ismethod
+
+
+class Job(Component, ABC):
```

Comment:
This name makes me think of HPC and workflow managers. What do you think about calling it `Task`?  This would bring us inline with [Luigi tasks](https://luigi.readthedocs.io/en/stable/tasks.html) which also define a `run` method. Though I am not sure if this is actually a good thing since we are _not_ building a workflow manager.